.circleci/config.yml

```diff
@@ -311,42 +339,88 @@ jobs:
 
   test-backend:
     executor: machine-executor
-    parameters:
-
-      owasp:
-        type: boolean
-        default: true
-
-      qasp:
-        type: boolean
-        default: true
-
-    branches:
     steps:
       - checkout
       - docker-compose-check
+      - docker-compose-up-backend
       - run:
-          name: "Test: Build and Spin-up Django API Service"
-          command: "cd tdrs-backend; docker-compose up -d --build"
-      - run:
-          name:  "Test: Execute Python Linting Test"
-          command: "cd tdrs-backend; docker-compose run --rm web bash -c \"flake8 .\""
+          name: Execute Python Linting Test
+          command: cd tdrs-backend; docker-compose run --rm web bash -c "flake8 ."
       - run:
-          name: "Test: Run Unit Tests And Create Code Coverage Report"
+          name: Run Unit Tests And Create Code Coverage Report
           command: |
             cd tdrs-backend;
             docker-compose run --rm web bash -c "./wait_for_services.sh && pytest --cov-report=xml"
       - upload-codecov:
           component: backend
           coverage-report: ./tdrs-backend/coverage.xml
+
+  test-frontend:
+    executor: machine-executor
+    working_directory: ~/tdp-apps
+    steps:
+      - checkout
+      - install-nodejs-machine
+      - node/install-yarn
+      - install-nodejs-packages:
+          app-dir: tdrs-frontend
+          cache-path: ~/.cache
+          cache-version: v2
+          pkg-manager: yarn
       - run:
-          name:  "Test: Execute Python OWASP ZAP Vulnerability Scan"
+          name: Run ESLint
+          command: cd tdrs-frontend; yarn lint
+      - run:
+          name: Run Pa11y Accessibility Tests
+          command: cd tdrs-frontend; mkdir pa11y-screenshots/; yarn test:accessibility
+      - run:
+          name: Run Jest Unit Tests
+          command: cd tdrs-frontend; yarn test:ci
+      - upload-codecov:
+          component: frontend
+          coverage-report: ./tdrs-frontend/coverage/lcov.info
+      - run:
+          name: Install lib bindings for Cypress
           command: |
-            if  << parameters.owasp >> || << parameters.qasp >> ; then
-              ./scripts/zap-scanner.sh backend circle
-            fi
+            sudo apt-get update
+            sudo apt-get install -y libgbm-dev
+      - run:
+          name: Run Cypress Integration Tests
+          command: cd tdrs-frontend; yarn cy:run:ci
       - store_artifacts:
-          path: tdrs-backend/reports/owasp_report.html
+          path: tdrs-frontend/pa11y-screenshots/
+
+  backend-owasp-scan:
+    executor: machine-executor
+    working_directory: ~/tdp-apps
+    steps:
+      - checkout
+      - docker-compose-check
+      - docker-compose-up-backend
+      - run:
+          name: Wait for Django to become available
```

Comment:
ZAP was attempting to start the scan before the backend was accessible, which caused failed builds. This will pause the build until the `zaproxy` container can reach the Django backend.